### PR TITLE
refactor(config): mark `keepalive` for advanced users

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Mark `keepalive` config option for advanced users
- Drop `experimental` status of `keepalive`

## Context

As suggested by `@rarkins` in:

- https://github.com/renovatebot/renovate/discussions/25726#discussioncomment-7564362

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/options/index.ts [hunk 1]
```diff
@@ -2357,7 +2357,7 @@ const options: RenovateOptions[] = [
     default: false,
```
Comment (HonkingGoose) [inline lib/config/options/index.ts:2357]: By the way the config option's name is `keepalive`, but we usually use camelcase. Should we rename the config option like this?

```diff
- name: 'keepalive',
+ name: 'keepAlive',
```

Comment (rarkins) [reply lib/config/options/index.ts:2357]: `got` appears to use camel case too: https://github.com/search?q=repo%3Asindresorhus%2Fgot%20keepalive&type=code

Comment (HonkingGoose) [reply lib/config/options/index.ts:2357]: Do you want an issue for renaming to `keepAlive` then?

Comment (rarkins) [reply lib/config/options/index.ts:2357]: @viceice do you recall why you named it with all lower case?

Comment (viceice) [reply lib/config/options/index.ts:2357]: don't remember, maybe I made a typo? need to check where it's passed to got

**Comment by rarkins** [comment]
Here's where it's used currently: https://github.com/search?q=repo%3Arenovatebot%2Frenovate%20keepaliveAgents&type=code

**Review by viceice — COMMENTED** [review]
so we should rename to caml case

**Comment by HonkingGoose** [comment]
> so we should rename to caml case

I'll let somebody else do this, so I don't get tangled up doing two things at once. See issue:

- #25816

**Review by rarkins — APPROVED** [review]

**Pull request merged**